chain_indexer_test failed

Hi, 

I run chain_indexer_test after I update my geth to 1.9.9 

#### System information

Geth version: 1.9.9
OS & Version: MacOSX

#### Expected behaviour
It should be PASS.

#### Actual behaviour
=== RUN   TestChainIndexerSingle
--- FAIL: TestChainIndexerSingle (5.79s)
    chain_indexer_test.go:155: Canonical section count mismatch: have 79, want 78

#### Try
````
func (b *testChainIndexBackend) reorg(headNum uint64) uint64 {
	firstChanged := (headNum + 1) / b.indexer.sectionSize
	if firstChanged < b.stored {
		b.stored = firstChanged
	}
	return b.stored * b.indexer.sectionSize
}
````
I found it can be PASS after I modified
```
 firstChanged := headNum / b.indexer.sectionSize to headNum + 1
```
to 
```
 firstChanged := (headNum + 1) / b.indexer.sectionSize to headNum + 1
```
in chain_indexer_test.go len 206.

Please suggest. Thank you.
@zsfelfoldi

--- a/core/chain_indexer_test.go
+++ b/core/chain_indexer_test.go
@@ -203,7 +203,7 @@ func (b *testChainIndexBackend) assertBlocks(headNum, failNum uint64) (uint64, b
 }
 
 func (b *testChainIndexBackend) reorg(headNum uint64) uint64 {
-	firstChanged := headNum / b.indexer.sectionSize
+	firstChanged := (headNum + 1) / b.indexer.sectionSize
 	if firstChanged < b.stored {
 		b.stored = firstChanged
 	}

